includes/classes/shopping_cart.php

```diff
@@ -1171,7 +1171,8 @@ function get_products($check_for_valid_cart = false) {
                            AND pd.products_id = p.products_id
                            AND pd.language_id = " . (int)$_SESSION['languages_id'] . " LIMIT 1";
 
-      if ($products = $db->Execute($products_query)) {
+      $products = $db->Execute($products_query);
+      if ($products->RecordCount()) {
```

Comment:
Sorry to be pedantic, but should that be either
```php
    if ($products->RecordCount() !== 0) {
```
or
```php
    if (!$products->EOF) {
```

Reply:
`      if ($products->RecordCount()) {`
is the same as 
`    if ($products->RecordCount() !== 0) {`

Reply:
Probably seen a reason before, but wondering why either alternative is considered necessary? Is it just "forward thinking" in that a "count" would expect a whole number or to be at the end of the listing?  Why though use `===` if the desired goal would be to not execute the internal code if it returned `false` or something `falsy`?

Reply:
I believe @lat9 is simply seeing it through the lens of strict types.
While the test for `falsy` here is perfectly functional, I'll concede that there's an argument to be made for clear indication of intent, so that the future reader doesn't have to wonder what was meant by what was written.
Thus in this case I will switch to testing for `EOF`.

(I don't like using `!== 0` or `=== 0` as I feel they convey the worry that it might ever not be numeric, which actually creates more confusion when reading code later, just like @mc12345678 asked about whether it's for "forward thinking" concerns.)

Reply:
Changed to `->EOF`